engine/engine_test.go

```diff
@@ -1975,6 +1975,13 @@ func TestInstantQuery(t *testing.T) {
 						http_requests_total{pod="nginx-2"} 1+2x4`,
 			query: "sum by (pod) (http_requests_total)",
 		},
+		{
+			name: "sum with NaN",
+			load: `load 30s
+						http_requests_total{pod="nginx-1"} 1+1x4
+						http_requests_total{pod="nginx-2"} NaN`,
+			query: "sum(http_requests_total)",
```

Comment:
Can we also add `sum by` with NaN since there's a different code path for grouping aggregates.

Reply:
added a test for `sum by` with NaN